dbms/src/Columns/ColumnString.h

```diff
@@ -333,8 +273,21 @@ class ColumnString final : public COWPtrHelper<IColumn, ColumnString>
         const NullMap * nullmap,
         const TiDB::TiDBCollatorPtr & collator,
         String * sort_key_container) const override;
-    void serializeToPos(PaddedPODArray<char *> & pos, size_t start, size_t length, bool has_null) const override;
+    template <bool has_null, bool need_decode_collator, typename DerivedCollator, bool has_nullmap>
+    void serializeToPosImpl(
```

Comment:
why make it public?

Reply:
Grouping all related function definitions enhances clarity and consistency. This approach aligns with functions like `insertDataImpl`, `compareAtWithCollationImpl`, `getPermutationWithCollationImpl`, and `updateWeakHash32Impl`, all of which are public.